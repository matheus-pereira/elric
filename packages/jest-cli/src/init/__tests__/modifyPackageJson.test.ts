/**
 * Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import modifyPackageJson from '../modifyPackageJson';

test('should remove elric config if exists', () => {
  expect(
    modifyPackageJson({
      projectPackageJson: {
        elric: {
          collectCoverage: true,
        },
      },
      shouldModifyScripts: true,
    }),
  ).toMatchSnapshot();
});

test('should add test script when there are no scripts', () => {
  expect(
    modifyPackageJson({
      projectPackageJson: {},
      shouldModifyScripts: true,
    }),
  ).toMatchSnapshot();
});

test('should add test script when there are scripts', () => {
  expect(
    modifyPackageJson({
      projectPackageJson: {
        scripts: {
          lint: 'eslint .',
          test: 'jasmine',
        },
      },
      shouldModifyScripts: true,
    }),
  ).toMatchSnapshot();
});

test('should not add test script when { shouldModifyScripts: false }', () => {
  expect(
    modifyPackageJson({
      projectPackageJson: {},
      shouldModifyScripts: false,
    }),
  ).toMatchSnapshot();
});
