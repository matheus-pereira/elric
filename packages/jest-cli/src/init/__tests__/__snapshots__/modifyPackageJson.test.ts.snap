// elric Snapshot v1, https://goo.gl/fbAQLP

exports[`should add test script when there are no scripts 1`] = `
"{
  \\"scripts\\": {
    \\"test\\": \\"elric\\"
  }
}
"
`;

exports[`should add test script when there are scripts 1`] = `
"{
  \\"scripts\\": {
    \\"lint\\": \\"eslint .\\",
    \\"test\\": \\"elric\\"
  }
}
"
`;

exports[`should not add test script when { shouldModifyScripts: false } 1`] = `
"{}
"
`;

exports[`should remove elric config if exists 1`] = `
"{
  \\"scripts\\": {
    \\"test\\": \\"elric\\"
  }
}
"
`;
